x-pack/plugins/security/server/lib/authentication/providers/saml.ts

```diff
@@ -518,25 +497,104 @@ export class SAMLAuthenticationProvider {
         { body: { acs: this.getACS() } }
       );
 
-      this.options.log(
-        ['debug', 'security', 'saml'],
-        'Redirecting to Identity Provider with SAML request.'
-      );
+      this.debug('Redirecting to Identity Provider with SAML request.');
 
       return AuthenticationResult.redirectTo(
         redirect,
         // Store request id in the state so that we can reuse it once we receive `SAMLResponse`.
         { requestId, nextURL: `${request.getBasePath()}${request.url.path}` }
       );
     } catch (err) {
-      this.options.log(
-        ['debug', 'security', 'saml'],
-        `Failed to initiate SAML handshake: ${err.message}`
-      );
+      this.debug(`Failed to initiate SAML handshake: ${err.message}`);
       return AuthenticationResult.failed(err);
     }
   }
 
+  /**
+   * Invalidates access and refresh tokens without calling `saml/logout`.
+   * @param accessToken Access token to invalidate.
+   * @param refreshToken Refresh token to invalidate.
+   */
+  private async performIdPInitiatedLocalLogout(accessToken: string, refreshToken: string) {
+    this.debug('Local logout has been initiated by the Identity Provider.');
+
+    // First invalidate old access token.
+    const {
```

Comment:
__note:__ we do the same for token provider (copied from there), we'll need to unify that at some point.